### PR TITLE
fix outstanding issues after #1315

### PR DESCRIPTION
**Pull request by hmmr**
Fix outstanding dialyzer issues after https://github.com/basho/riak_kv/pull/1315.

Depends on https://github.com/basho/riak_ql/pull/80. A r_t module checking some window functions is https://github.com/basho/riak_test/pull/966.

Andalso, fix value returned by `riak_kv_pb_timeseries:get_column_types/2`.

### CODE REVIEW

File: src/riak_kv_qry_compiler.erl [hunk 1]
```diff
@@ -156,7 +156,7 @@ select_column_clause_folder(DDL, ColX, {TypeSet1, SelClause1}) ->
         col_return_types = lists:flatten(ColRet2),
         clause           = RunFn2,
         is_valid         = IsValid2,
-        finalisers       = lists:flatten(Finalisers2)},
+        finalisers       = Finalisers2},
```
Comment (gordonguthrie) [inline src/riak_kv_qry_compiler.erl]: This will (probably) fail with a select \* clause

File: src/riak_kv_qry_compiler.erl [hunk 2]
```diff
@@ -210,24 +211,23 @@ compile_select_col(DDL, {{window_agg_fn, FnName}, [FnArg1]}) ->
...
     %% arity two, when we know the function is stateless wrap it in a two arity
     %% fun and ignore the state arg
-    Finalisers = lists:duplicate(length(ColTypes), fun(State) -> State end),
     #single_sel_column{ calc_type        = rows,
                         initial_state    = undefined,
                         col_return_types = ColTypes,
                         clause           = fun(Row, _) -> Fn(Row) end,
                         is_valid         = IsValid,
-                        finaliser        = Finalisers }.
+                        finaliser        = fun(X) -> X end }.
```
Comment (gordonguthrie) [inline src/riak_kv_qry_compiler.erl]: Not sure this is right either?

**Comment by hazen** [comment]
@hmmr LOL. You are doing exactly what I was doing.  I was going to test your branch, but needed to merge in `e2e/ts`.  Since it was only 4:00am in the Ukraine, I thought you **might** be asleep.

**Comment by andytill** [comment]
+1  777212a

**Pull request merged**